target/linux/lantiq/files/arch/mips/boot/dts/lantiq/vr9_avm_fritz7490.dts

```diff
@@ -0,0 +1,282 @@
+		};
+		// led 2 "WLAN"
+		led_wlan: wlan {
+			label = "fritz7490:green:wlan";
+			gpios = <&gpio 35 GPIO_ACTIVE_LOW>;
+		};
+		// led 1 "Info"
+		led_info: info {
+			label = "fritz7490:green:info";
+			gpios = <&gpio 33 GPIO_ACTIVE_LOW>;
+		};
+		led_info_red: info_red {
+			label = "fritz7490:red:info";
+			gpios = <&gpio 46 GPIO_ACTIVE_LOW>;
+		};
+	};
+
+	gpio-export {
+		compatible = "gpio-export";
+
+		gpio_wasp_reset {
+			gpio-export,name = "fritz7490:wasp:reset";
+			gpio-export,output = <1>;
+			gpios = <&gpio 34 GPIO_ACTIVE_LOW>;
+		};
+		gpio_wasp_wakeup {
+			gpio-export,name = "fritz7490:wasp:wakeup";
+			gpio-export,output = <2>;
+			gpios = <&gpio 5 GPIO_ACTIVE_HIGH>;
+		};
+	};
+};
+
+&eth0 {
+	interface@0 {
+		compatible = "lantiq,xrx200-pdi";
+		#address-cells = <1>;
+		#size-cells = <0>;
+		reg = <0>;
+		lantiq,switch;
+
+		ethernet@0 {
+			compatible = "lantiq,xrx200-pdi-port";
+			reg = <0>;
+			phy-mode = "sgmii";
+			phy-handle = <&phy0>;
+			gpios = <&gpio 32 GPIO_ACTIVE_HIGH>;
+		};
+
+		ethernet@1 {
+			compatible = "lantiq,xrx200-pdi-port";
+			reg = <1>;
+			phy-mode = "sgmii";
+			phy-handle = <&phy1>;
+			gpios = <&gpio 44 GPIO_ACTIVE_HIGH>;
+		};
+
+		ethernet@2 {
+			compatible = "lantiq,xrx200-pdi-port";
+			reg = <2>;
+			phy-mode = "gmii";
+			phy-handle = <&phy11>;
+		};
+
+		ethernet@4 {
+			compatible = "lantiq,xrx200-pdi-port";
+			reg = <4>;
+			phy-mode = "gmii";
+			phy-handle = <&phy13>;
+		};
+
+		ethernet@5 {
+			compatible = "lantiq,xrx200-pdi-port";
+			reg = <5>;
+			phy-mode = "rgmii";
+
+			fixed-link {
+				speed = <1000>;
+				full-duplex;
+			};
+		};
+	};
+
+	mdio {
+		#address-cells = <1>;
+		#size-cells = <0>;
+		compatible = "lantiq,xrx200-mdio";
+
+		phy0: ethernet-phy@0 {
+			reg = <0x0>;
+			compatible = "ethernet-phy-ieee802.3-c22";
+		};
+
+		phy1: ethernet-phy@1 {
+			reg = <0x1>;
+			compatible = "ethernet-phy-ieee802.3-c22";
+		};
+
+		phy11: ethernet-phy@11 {
+			reg = <0x11>;
+			compatible = "lantiq,phy11g", "ethernet-phy-ieee802.3-c22";
+		};
+
+		phy13: ethernet-phy@13 {
+			reg = <0x13>;
+			compatible = "lantiq,phy11g", "ethernet-phy-ieee802.3-c22";
+		};
+	};
+};
+
+&gphy0 {
+	lantiq,gphy-mode = <GPHY_MODE_GE>;
+};
+
+&gphy1 {
+	lantiq,gphy-mode = <GPHY_MODE_GE>;
+};
+
+&gpio {
+	pinctrl-names = "default";
+	pinctrl-0 = <&state_default>;
+
+	state_default: pinmux {
+		phy-rst {
+			lantiq,pins = "io32", "io44";
+			lantiq,pull = <0>;
+			lantiq,open-drain;
+			lantiq,output = <1>;
+		};
+
+		pcie-rst {
+			lantiq,pins = "io21";
+			lantiq,open-drain;
+			lantiq,output = <1>;
+		};
+	};
+
+	pcie-rst-dev {
+		gpio-hog;
+		line-name = "pcie-rst-dev";
+		gpios = <22 GPIO_ACTIVE_LOW>;
+		output-low;
+	};
+
+	usb-vbus {
+		gpio-hog;
+		line-name = "usb-vbus";
+		gpios = <14 GPIO_ACTIVE_HIGH>;
+		output-low;
+	};
+};
+
+&spi {
+	status = "okay";
+
+	flash@4 {
+		compatible = "jedec,spi-nor";
+		reg = <4>;
+		spi-max-frequency = <10000000>;
```

Comment:
Faster possible?

Reply:
I will check the datasheet...

Reply:
Nand is Micron MT29F4G08ABADAWP

Available in 100-166 MT/s

Maybe 80000000 Hz to be on the safe side?
Not sure for reference:
``amazonse_netgear_dgn1000b.dts:`		spi-max-frequency = <5000000>;
ar9_netgear_dgn3500.dtsi:		spi-max-frequency = <20000000>;
falcon_lantiq_easy98000.dtsi:		spi-max-frequency = <1000000>;
falcon_sflash-16m.dtsi:		spi-max-frequency = <80000000>;
vr9_avm_fritz3370-rev2.dtsi:		spi-max-frequency = <1000000>;
vr9_avm_fritz7362sl.dts:		spi-max-frequency = <1000000>;
vr9_buffalo_wbmr-300hpd.dts:		spi-max-frequency = <20000000>;
vr9_lantiq_easy80920.dtsi:		spi-max-frequency = <1000000>;
vr9_netgear_dm200.dts:		spi-max-frequency = <10000000>;
vr9_tplink_tdw89x0.dtsi:		spi-max-frequency = <33250000>;
``

Reply:
80MHz is to fast.. getting timeouts. 40 Mhz is doing fine.

Cannot find anything in the original sources...
Going for  33.25 Mhz

Reply:
I went down to 10Mhz on the 3390 to be on the safe side. I experienced the same timeouts as you did, but I don't remember at which setting.
IMHO it's not very critical on devices with NAND, since SPI transfers only happen during reading of EEPROM data and MAC addresses.